spikeinterface/extractors/stream_ibl.py

```diff
@@ -0,0 +1,239 @@
+        # when this happens a statement is printed to stderr saying these are using default metadata configurations
+        with redirect_stderr(StringIO()):
+            electrodes_geometry = self._file_streamer.geometry
+
+        if not load_sync_channel:
+            shank = electrodes_geometry["shank"]
+            shank_row = electrodes_geometry["row"]
+            shank_col = electrodes_geometry["col"]
+            inter_sample_shift = electrodes_geometry["sample_shift"]
+            adc = electrodes_geometry["adc"]
+            index_on_probe = electrodes_geometry["ind"]
+            good_channel = electrodes_geometry["flag"]
+        else:
+            shank = np.concatenate((electrodes_geometry["shank"], [np.nan]))
+            shank_row = np.concatenate((electrodes_geometry["shank"], [np.nan]))
+            shank_col = np.concatenate((electrodes_geometry["shank"], [np.nan]))
+            inter_sample_shift = np.concatenate((electrodes_geometry["sample_shift"], [np.nan]))
+            adc = np.concatenate((electrodes_geometry["adc"], [np.nan]))
+            index_on_probe = np.concatenate((electrodes_geometry["ind"], [np.nan]))
+            good_channel = np.concatenate((electrodes_geometry["shank"], [1.]))
+
+        self.set_property("shank", shank)
+        self.set_property("shank_row", shank_row)
+        self.set_property("shank_col", shank_col)
+        self.set_property("inter_sample_shift", inter_sample_shift)
+        self.set_property("adc", adc)
+        self.set_property("index_on_probe", index_on_probe)
+        if not all(good_channel):
+            self.set_property("good_channel", good_channel)
+
+        # init recording segment
+        recording_segment = StreamingIblRecordingSegment(
+            file_streamer=self._file_streamer,
+            load_sync_channel=load_sync_channel
+        )
+        self.add_recording_segment(recording_segment)
+
+        self._kwargs = {
+            "session": session,
+            "stream_name": stream_name,
+            "load_sync_channel": load_sync_channel,
+            "cache_folder": cache_folder,
+            "remove_cached": remove_cached,
+        }
+
+
+class StreamingIblRecordingSegment(BaseRecordingSegment):
+    def __init__(self, file_streamer, load_sync_channel: bool = False):
+        BaseRecordingSegment.__init__(self, sampling_frequency=file_streamer.fs)
+        self._file_streamer = file_streamer
+        self._load_sync_channel = load_sync_channel
+
+    def get_num_samples(self):
+        return self._file_streamer.ns
+
+    def get_traces(self, start_frame: int, end_frame: int, channel_indices):
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self.get_num_samples()
+        if channel_indices is None:
+            channel_indices = slice(None)
+
+        traces = np.rint(self._file_streamer[start_frame:end_frame] / self._file_streamer.sample2volts).astype("int16")
```

Comment:
@alejoe91 I also just noticed some casting errors without calling some sort of rounding before `int16` casting

![image](https://user-images.githubusercontent.com/51133164/214622530-bb50f70e-a154-4212-8538-ae6ea2f71fe2.png)

Reply:
Sounds good! Thanks for catching it